src/fights.ts

```diff
@@ -361,20 +358,28 @@ function pygmyOptions(forceEquip: Item[] = []) {
 let _bestShadowRift: Location | null = null;
 export function bestShadowRift(): Location {
   if (!_bestShadowRift) {
-    _bestShadowRift =
-      chooseRift({
-        canAdventure: true,
-        sortBy: (l: Location) => {
-          const drops = getMonsters(l)
-            .map((m) => Object.keys(itemDrops(m)).map((s) => toItem(s)))
-            .flat();
-          return sum(drops, garboValue);
-        },
-      }) ?? $location.none;
+    _bestShadowRift = ClosedCircuitPayphone.chooseRift({
+      canAdventure: true,
+      sortBy: (l: Location) => {
+        setLocation(l);
+        // We probably aren't capping item drops with the penalty
+        // so we don't really need to compute the actual outfit (or the dropModifier for that matter actually)
+        const dropModifier = 1 + numericModifier("Item Drop") / 100;
+        return sum(getMonsters(l), (m) => {
+          const monsterDrops = itemDrops(m);
+          const items = Object.keys(monsterDrops).map(toItem);
+          const rates = Object.values(monsterDrops);
+          return sumNumbers(
+            items.map((item, idx) => garboValue(item) * clamp(rates[idx] * dropModifier, 0, 1))
+          );
```

Comment:
it won't let me make a formal suggestion for whatever reason, which sucks, but:
```
const drops = [...Object.entries(monsterDrops)];
return sum(drops, ([drop, rate]) => garboValue(toItem(drop)) * clamp(rate * dropModifier / 100, 0, 1));
```
alternately, and this might be easier:
```
return sum(itemDropsArray(m), ({ drop, rate }) => garboValue(drop) * clamp(rate * dropModifier / 100, 0, 1));
```

okay, yeah, definitely the second one

Reply:
`drop` is an Item in `itemDropsArray` but a string in `itemDrops` because only strings can be the keys to an object, but itemDropsArray doesn't actually use the drop as a key